adapters/uniqueid_linux.c

```diff
@@ -25,13 +39,38 @@ UNIQUEID_RESULT UniqueId_Generate(char* uid, size_t len)
     }
     else
     {
-        uuid_t uuidVal;
-        uuid_generate(uuidVal);
 
-        /* Codes_SRS_UNIQUEID_07_001: [UniqueId_Generate shall create a unique Id 36 character long string.] */
-        memset(uid, 0, len);
-        uuid_unparse(uuidVal, uid);
-        result = UNIQUEID_OK;
+        // generate version 4 uuid
+        UUID unique_id;
+
+        // get random bytes
+        ssize_t bytes_got = getrandom(&unique_id, sizeof(unique_id), GRND_NONBLOCK);
+        if (bytes_got != sizeof(unique_id)) {
```

Comment:
Can you also specify the case where -1 could be returned?  (I get that size of unique_id won't be this, but I think it is more clear)